bin.src/genDefectList.py

```diff
@@ -0,0 +1,265 @@
+    dataDir = os.path.join(os.path.join(args.root, 'rerun'), args.rerun)
+butler = dafPersist.Butler( dataDir )
+
+biasesVisit = list()
+flatsVisit = list()
+flatsExpTimes = list()
+flatsLow = list()
+flatsHigh = list()
+for visit in np.arange( args.visitLow, args.visitHigh + 1 ):
+    try:
+        dataId = dict( spectrograph=args.spectrograph, arm=args.arm, visit=visit )
+        fileName = butler.get( 'raw_filename', dataId )
+        md = afwImage.readMetadata( fileName[ 0 ] )
+
+        if md.get("IMAGETYP") == "bias":
+            biasesVisit.append(visit)
+        if md.get("IMAGETYP") == "flat":
+            flatsVisit.append(visit)
+            flatsExpTimes.append(md.get('EXPTIME'))
+            if md.get('EXPTIME') == args.expTimeLow:
+                flatsLow.append(visit)
+            elif md.get('EXPTIME') == args.expTimeHigh:
+                flatsHigh.append(visit)
+    except:
+        logger.debug("visit %d not found" % visit)
+
+logger.debug("len(flatsLow) = %d" % len(flatsLow))
+logger.debug("flatsLow[0] = %d" % flatsLow[0])
+
+dataId = dict(visit=flatsLow[0], arm=args.arm, spectrograph=args.spectrograph)
+exp = butler.get('postISRCCD', dataId, immediate=True)
+image = exp.getMaskedImage().getImage().getArray()
+ccd = exp.getDetector().getId()
+logger.info("detector Id = %d" % ccd)
+
+if textArrRead:
+    firstWords = []
+    for i in range(len(textArr)):
+        firstWords.append(textArr[i].split()[0])
+    logger.debug('firstWords = %s' % firstWords)
+    for i in np.arange(len(firstWords)-1,-1,-1):
+        if firstWords[i][0] == '#':
+            textArr.pop(i)
+            firstWords.pop(i)
+    logger.debug('firstWords = %s' % firstWords)
+    uniqueFirstInts = [int(x) for x in np.unique(firstWords)]
+    uniqueFirstIntsSorted = np.sort(uniqueFirstInts)
+    firstInts = [int(x) for x in firstWords]
+    logger.debug('uniqueFirstIntsSorted = %s' % uniqueFirstIntsSorted)
+    for a in uniqueFirstIntsSorted:
+        if a < ccd:
+            for i in range(len(textArr)):
+                if firstInts[i] == a:
+                    text_file.write(textArr[i])
+
+allFlatsLow = np.ndarray(shape=[image.shape[0], image.shape[1], len(flatsLow)], dtype='float32')
+allFlatsHigh = np.ndarray(shape=[image.shape[0], image.shape[1], len(flatsHigh)], dtype='float32')
+for i in np.arange(len(flatsLow)):
+    dataId = dict(visit=flatsLow[i], spectrograph=args.spectrograph, arm=args.arm)
+    image = butler.get('postISRCCD', dataId).getMaskedImage().getImage().getArray()
+    allFlatsLow[:,:,i] = image
+for i in np.arange(len(flatsHigh)):
+    dataId = dict(visit=flatsHigh[i], spectrograph=args.spectrograph, arm=args.arm)
+    image = butler.get('postISRCCD', dataId).getMaskedImage().getImage().getArray()
+    allFlatsHigh[:,:,i] = image
+
+nLow = len(flatsLow)
+if nLow > 2:
+    medianFlatsLow = np.median(allFlatsLow,2)
+    logger.info('%d Flats with %ds exposure time found' % (nLow, args.expTimeLow))
+elif nLow == 2:
```

Comment:
It's a starting point if we don't have any more flats. A warning is given out if there are less than 3 flats for any of the 2 exposure times.